nightingale-server-prod.yaml

```diff
@@ -0,0 +1,198 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: nightingale
+---
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: default-mem-limits
+  namespace: nightingale
+spec:
+  limits:
+  - default:
+      memory: 512Mi
+    defaultRequest:
+      memory: 512Mi
+    type: Container
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nightingale
+  namespace: nightingale
+  annotations:
+    flux.weave.works/automated: "true"
+    flux.weave.works/tag.nightingale-server: glob:master-*
+    flux.weave.works/tag.migrate: glob:master-*
+    reloader.stakater.com/auto: "true"
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: nightingale
+  template:
+    metadata:
+      labels:
+        app: nightingale
+    spec:
+      containers:
+      - name: nightingale
+        image: harbor.nightingale.revelry.org/library/nightingale-server:develop-c030c6e
+        ports:
+        - containerPort: 5000
+        envFrom:
+        - configMapRef:
+            name: nightingale-env-config
+        - secretRef:
+            name: nightingale-env-secret
+        - secretRef:
+            name: kube-pull
```

Comment:
I'm sure this is wrong but not sure how yet. `kube-pull` is an image pull secret, not something the container itself needs access to. But I'm not sure if it has to be actually referenced or if it gets injected by Flux. Checking...

Reply:
`imagePullSecrets` is an array you can specify at the pod level (even with `containers` and `initContainers` rather than inside them).

Reply:
Yeah let's do that. (Apparently in revelry-prod we do something different, but I don't love the opaqueness of the alternative method.)

Reply:
FWIW the _other_ way to do it is to add it to the default service account in the namespace (or to a specific service account attached to the pod) but both of those ways are uglier IMHO.

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account

Reply:
@CurtisVonRubenhoff This is what you should be doing instead:

```yaml
spec:
  imagePullSecrets:
    - name: kube-pull
  containers:
    ...
```

Reply:
got it